### ISSUE TITLE
Menu cannot be disabled

### ISSUE THREAD

**Issue by ioanabrooks** [aws-amplify/amplify-ui#5164]
### Before creating a new issue, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-ui/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-ui/discussions).
- [X] I have tried disabling all browser extensions or using a different browser
- [X] I have tried deleting the node_modules folder and reinstalling my dependencies
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md#bug-reports).

### On which framework/platform are you having an issue?

React

### Which UI component?

Primitive components

### How is your app built?

CRA

### What browsers are you seeing the problem on?

Chrome, Firefox, Safari

### Please describe your bug.

MenuButton component cannot be disabled, `isDisabled` prop has no effect. `disabled` prop gets propagated to the button and updates styles, but it is still clickable. 

### What's the expected behaviour?

Should be able to disable the Menu button through isDisabled prop.

### Help us reproduce the bug!

CodeSandbox link: https://codesandbox.io/p/sandbox/auth-styling-forked-zv7ykv?file=%2Fsrc%2FApp.tsx

* using latest version of ui-react

### Code Snippet

```typescript
// Put your code below this line.

 <MenuButton variation="primary" size="large" isDisabled>
            With isDisabled prop
 </MenuButton>

 <MenuButton variation="primary" size="large" disabled>
            With isDisabled prop
 </MenuButton>


```

**Comment by esauerbo** [comment]
I was able to reproduce this with the latest `@aws-amplify/ui-react` and am looking into a fix.

**Comment by esauerbo** [comment]
Discussed this with the team and disabling the `MenuButton` alone should not actually disable the `Menu`, as this would require passing props up to the `DropdownMenuTrigger` which is against React standards. Instead, the `isDisabled` prop should be passed to the `Menu` itself, i.e.
```
<Menu isDisabled />
```
https://github.com/aws-amplify/amplify-ui/blob/b821d4ae69c5eb4b5971cdc211e206e6f23698f7/packages/react/src/primitives/Menu/Menu.tsx#L45

Will leave this open to a track a fix on correctly disabling the `Menu` component.

**Comment by esauerbo** [comment]
This has been fixed in `@aws-amplify/ui-react@6.1.8` https://www.npmjs.com/package/@aws-amplify/ui-react?activeTab=versions
